<p>This rule checks that common layer does not depend on another layer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: blue'>common</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: red'>client</span>.api.Foo;
    import com.devonfw.ide.sonarqube.<span style='color: red'>service</span>.api.Bar;
    import com.devonfw.ide.sonarqube.<span style='color: red'>logic</span>.api.Some;
    
    public class MyClass {}
</pre>