<p>
    This rule checks that client layer does not depend on logic layer.
    Allowed dependencies from client layer:
    <ul type='disc'>
        <li>Client layer of another component</li>
        <li>Service layer of the same component</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: blue'>client</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: red'>logic</span>.api.Bar;
    
    public class MyClass {}
</pre>