<p>This rule checks that classes inheriting from Cto classes end with the suffix Cto.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: red'>MyClass</span> extends AbstractCto {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: green'>ClassCto</span> extends AbstractCto {}
</pre>