<p>This rule checks that classes inheriting from entity classes end with the suffix Entity.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: red'>TestClass</span> extends SuperEntity {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: green'>TestEntity</span> extends SuperEntity {}
</pre>