<p>This rule checks that classes inheriting from classes with the suffix Eto end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: red'>TestClass</span> extends AbstractEto {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: green'>TestEto</span> extends AbstractEto {}
</pre>