<p>This rule checks that classes extending classes with the suffix To end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: red'>TestClass</span> extends AbstractTo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: green'>UserTo</span> extends AbstractTo {}
</pre>