<p>
    This rule checks that api scope does not depend on impl scope.
    Api scope is only allowed to depend on the api scope of another component.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.common.<span style='color: blue'>api</span>;

    import com.devonfw.ide.sonarqube.common.<span style='color: red'>impl</span>.Bar;
    
    public class MyClass {}
</pre>