<p>
    This rule checks that base scope does not depend on base scope of another component part.
    Allowed dependencies from base scope:
    <ul type='disc'>
        <li>Api scope of the same component</li>
        <li>Api scope of a different component</li>
        <li>Base scope of a different component</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: blue'>component1.logic.base</span>;

    import com.devonfw.ide.sonarqube.<span style='color: red'>component2.logic.base</span>.Bar;

    public class MyClass {}
</pre>