<p>
    This rule checks that the dataaccess layers of different components don't depend on each other. 
    From the dataaccess layer, only dependencies to the common layer are allowed.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: blue'>compA.dataaccess</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: red'>compB.dataaccess</span>.api.Bar;

    public class MyClass {}
</pre>