<p>This rule checks that all components only depend on components that were declared as dependencies in the architecture.json</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>