<p>This rule checks that the service layer of one component does not depend on the logic layer of another component</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>