<p>This rule checks that the logic layer of one component does not depend on the dataaccess layer of another component</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>