<p>This rule checks that non-abstract classes extending AbstractDao end with DaoImpl. In addition, classes that inherit from a class with the suffix Dao must also have the suffix Dao</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>