<p>This rule checks that non-abstract classes inheriting from AbstractUc begin with Uc, end with Impl and implement an interface with the same name except the suffix Impl</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>