<p>This rule checks that classes inheriting from classes with the suffix SearchCriteriaTo end with the same suffix</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>