<p>This rule checks that interfaces inheriting from an interface with the suffix Dao end with the same suffix</p>
<h2>Noncompliant Code Example</h2>
<pre></pre>
<h2>Compliant Solution</h2>
<pre></pre>